From 20bb45563fe8f3ec95ef22d715d1add014156543 Mon Sep 17 00:00:00 2001
From: Quirin Gylstorff <quirin.gylstorff@siemens.com>
Date: Wed, 29 Sep 2021 15:28:21 +0200
Subject: [PATCH 1/7] debian/config: Make image encryption optional

This can be use to ease the setup with SWUpdate.

Signed-off-by: Quirin Gylstorff <quirin.gylstorff@siemens.com>
---
 debian/configs/defconfig | 1 -
 debian/rules             | 3 +++
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/debian/configs/defconfig b/debian/configs/defconfig
index 02681e53..b34168e3 100644
--- a/debian/configs/defconfig
+++ b/debian/configs/defconfig
@@ -3,7 +3,6 @@ CONFIG_HW_COMPATIBILITY=y
 CONFIG_DOWNLOAD=y
 CONFIG_DOWNLOAD_SSL=y
 CONFIG_SIGALG_CMS=y
-CONFIG_ENCRYPTED_IMAGES=y
 CONFIG_SURICATTA=y
 CONFIG_SURICATTA_SSL=y
 CONFIG_UPDATE_STATE_CHOICE_BOOTLOADER=y
diff --git a/debian/rules b/debian/rules
index 864add23..08b74a1d 100755
--- a/debian/rules
+++ b/debian/rules
@@ -41,6 +41,9 @@ endif
 ifeq (,$(filter pkg.swupdate.nosigning,$(DEB_BUILD_PROFILES)))
 	echo CONFIG_SIGNED_IMAGES=y >> configs/debian_defconfig
 endif
+ifeq (,$(filter pkg.swupdate.noencryption,$(DEB_BUILD_PROFILES)))
+	echo CONFIG_ENCRYPTED_IMAGES=y >> configs/debian_defconfig
+endif
 ifneq (,$(filter pkg.swupdate.p11,$(DEB_BUILD_PROFILES)))
 	echo CONFIG_PKCS11=y >> configs/debian_defconfig
 endif
-- 
2.34.1

